erpnext/erpnext_integrations/doctype/affirm_settings/affirm_settings.py

```diff
@@ -223,8 +223,25 @@ def authorize_payment(affirm_data, reference_doctype, reference_docname, integra
 	"""
 		once callback return checkout token it will authroized payment status as failed or sucessful
 	"""
+	payment_request = frappe.get_doc(reference_doctype, reference_docname)
 	redirect_url = "/integrations/payment-failed"
 
+	#if the payment request generated from a quotation, we convert quotation to sale order
+	if payment_request.reference_doctype == "Quotation" and not affirm_data.get("type") == "invalid_request":
+		
+		#we get the quotation document and submit it
+		quotation = frappe.get_doc(payment_request.reference_doctype, payment_request.reference_name)
+		quotation.save(ignore_permissions=True)
+		quotation.submit()
+
+		#convert the quotation to a sales order
+		from erpnext.selling.doctype.quotation.quotation import _make_sales_order
+		sales_order = frappe.get_doc(_make_sales_order(quotation.name, ignore_permissions=True))
+		sales_order.payment_schedule = []
+		sales_order.flags.ignore_permissions = True
+		sales_order.insert()
+		sales_order.save()
+
```

Comment:
@rvpasari can you also add a comment here that payment entry is made when the payment is captured ?

Reply:
Will add those in the revisions I'm making